server/commands/VolumeCommand.ts

```diff
@@ -2,17 +2,40 @@ import CommandDefinition from '../domain/CommandDefinition';
 import IoConnection from '../clients/IoConnection';
 import VolumeEventMessage from "../domain/event-messages/VolumeEventMessage";
 
+const volChangeTypes: { [s: string]: string; } = {
+  '+': 'increase',
+  '-': 'decrease',
+  default: 'set',
+};
+
+function getVolChangeType(value: string):string {
+  return volChangeTypes[value.charAt(0)] || volChangeTypes.default;
+}
+
+function getFeedbackMsg(eventMsg:VolumeEventMessage) {
+  if (volChangeTypes['+'] === eventMsg.changeType) {
+    return `Zwiększono głośność o "${eventMsg.volume}"`;
+  }
+  if (volChangeTypes['-'] === eventMsg.changeType) {
+    return `Zmniejszono głośność o "${eventMsg.volume}"`;
+  }
+  return `Ustawiono głośność na "${eventMsg.volume}"`;
+}
+
 async function volume(value: string) : Promise<string> {
-  const parsedInt: number = parseInt(value, 10);
+  const parsedInt: number = Math.abs(parseInt(value, 10));
```

Comment:
Wouldn't it be simpler to pass positive or negative value and have two modes: relative and absolute?

Now you have three modes: setting, increasing and decreasing. Increasing and decreasing are very similar and come down to just adding positive or negative value to current volume.

This looks like simpler, less-code-required solution. Am I missing something?

Reply:
done